src/main/java/com/fasterxml/jackson/databind/deser/CreatorProperty.java

```diff
@@ -76,19 +77,40 @@ public class CreatorProperty
     protected boolean _ignorable;
 
     /**
-     * @since 2.11
+     * Marker flag to indicate that current property is used to handle "any setter" via `@JsonAnySetter`.
+     *
+     * @since 2.18
+     */
+    protected final Boolean _isAnySetterProp;
+
+    /**
+     * @since 2.18
      */
     protected CreatorProperty(PropertyName name, JavaType type, PropertyName wrapperName,
             TypeDeserializer typeDeser,
             Annotations contextAnnotations, AnnotatedParameter param,
             int index, JacksonInject.Value injectable,
-            PropertyMetadata metadata)
+            PropertyMetadata metadata, Boolean isAnySetterProp)
```

Comment:
At this point should probably be primitive `boolean` as there should not be any use for `null`?